src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtAuthenticationFilter.java

```diff
@@ -43,11 +50,32 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         filterChain.doFilter(request, response);
     }
 
+    private void addRefreshTokenToResponse(
+        HttpServletResponse response, String jwt, UsernamePasswordAuthenticationToken authentication
+    ) {
+        Date newDate = null;
+        try {
+            newDate = tokenProvider.getMaxExpiryDateFromJwt(jwt);
+        } catch (RuntimeException rex) {
+            logger.error("Could not find maxEpiryDate in jwt Token", rex);
+        }
+        response.addHeader(newTokenHeaderName, createJwtForResponse(
```

Comment:
I do not get this logic, so null will be added when an exception? why would we add the header at all if there is a problem with getting max expiry date